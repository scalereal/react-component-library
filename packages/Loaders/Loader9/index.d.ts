import * as React from 'react';
import { WithStyle } from '../../utils';
import { Props } from '../types';
declare const Loader9: React.FC<Props> & WithStyle;
export default Loader9;
