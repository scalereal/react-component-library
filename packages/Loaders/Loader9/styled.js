var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
import styled from 'styled-components';
export var Loader = styled.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n    display: inline-block;\n    width: 1em;\n    height: 1em;\n    color: ", ";\n    vertical-align: middle;\n    pointer-events: none;\n    \n    background-color: currentcolor;\n\tanimation: loader-11 1.2s infinite ease-in-out;\n\n    @keyframes loader-11 {\n        0% {\n            transform: perspective(120px) rotateX(0deg) rotateY(0deg);\n        }\n        50% {\n            transform: perspective(120px) rotateX(-180.1deg) rotateY(0deg);\n        }\n        100% {\n            transform: perspective(120px) rotateX(-180deg) rotateY(-179.9deg);\n        }\n    }\n\n"], ["\n    display: inline-block;\n    width: 1em;\n    height: 1em;\n    color: ", ";\n    vertical-align: middle;\n    pointer-events: none;\n    \n    background-color: currentcolor;\n\tanimation: loader-11 1.2s infinite ease-in-out;\n\n    @keyframes loader-11 {\n        0% {\n            transform: perspective(120px) rotateX(0deg) rotateY(0deg);\n        }\n        50% {\n            transform: perspective(120px) rotateX(-180.1deg) rotateY(0deg);\n        }\n        100% {\n            transform: perspective(120px) rotateX(-180deg) rotateY(-179.9deg);\n        }\n    }\n\n"])), function (pr) { return pr.Color ? pr.Color : 'inherit'; });
var mixin = function (size) {
    switch (size) {
        case "XXS":
            return "font-size: 0.8rem;\n           ";
        case "XS":
            return "font-size: 1rem;\n            \n            ";
        case "S":
            return "font-size: 2rem;\n            ";
        case "M":
            return "font-size: 3rem;\n            ";
        case "L":
            return "font-size: 5rem;\n            ";
        case "XL":
            return "font-size: 6rem;\n            ";
        default:
            return "font-size: 1rem;\n       ";
    }
};
export var LoaderWrapper = styled.div(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n    ", "\n    ", "\n    display: ", ";\n"], ["\n    ", "\n    ", "\n    display: ", ";\n"])), function (pr) { return pr.disabled === true ? 'pointer-events: none;opacity: 0.4;' : ''; }, function (pr) { return mixin(pr.size); }, function (pr) { return pr.visible ? 'inline-block' : 'none'; });
var templateObject_1, templateObject_2;
