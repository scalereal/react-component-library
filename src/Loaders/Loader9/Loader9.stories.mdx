import Loader9 from '.';
import { SizeMap } from '../types';
import { Preview, Story, Meta, Props, ArgsTable, Canvas } from '@storybook/addon-docs/blocks';
import { withKnobs, color, boolean, select } from '@storybook/addon-knobs';

<Meta
    title="Components/Loaders/Loader9"
    component={Loader9}
    argTypes={{
        size: {
            control: {
                type: 'select',
                options: Object.values(SizeMap)
            }
        },
        Color: {
            control: 'color'
        },
        display: {
            control: {
                type: 'boolean'
            }
        },
        ref: {
            table: {
                disable: true
            }
        },
        theme: {
            table: {
                disable: true
            }
        },
        as: {
            table: {
                disable: true
            }
        }
    }}
/>

# Loaders

Displays a Loader.

export const Template = args => (
    <div>
        <Loader9 {...args} />
    </div>
);

<Preview>
    <Story name="Loader9" args={{ size: 'S' }}>
        {Template.bind({})}
    </Story>
</Preview>

### Color Variations

You can either change the theme or pass the colors as props.

<Preview withToolbar>
    <div style={{ display: 'flex', justifyContent: 'space-around' }}>
        <Loader9 size="S" Color="green" />
        <Loader9 size="S" Color="red" />
    </div>
</Preview>

### Sizes

There are Six sizs available `XXS` | `XS` | `S` | `M` | `L` | `XL`.

<Preview withToolbar>
    <div style={{ display: 'flex', justifyContent: 'space-between' }}>
        <DotsCircle size="XXS" />
        <Loader9 size="XS" />
        <Loader9 size="S" />
        <Loader9 size="M" />
        <Loader9 size="L" />
        <Loader9 size="XL" />
    </div>
</Preview>

### Props

<Props story="Loader9" />
