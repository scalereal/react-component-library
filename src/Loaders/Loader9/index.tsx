import * as React from 'react';
import { WithStyle } from '../../utils';
import { Props } from '../types';
import { Loader, LoaderWrapper } from './styled';

const Loader9:React.FC<Props> & WithStyle = React.memo(React.forwardRef( (props, ref) => {
    const {size} = props;

    return (
        <LoaderWrapper ref={ref} size={size} {...props}>
            <Loader {...props}/>
        </LoaderWrapper>
    );
    
}));

Loader9.displayName="Loader9";
Loader9.defaultProps = {
    size:'S',
    visible: true
};

export default Loader9;
