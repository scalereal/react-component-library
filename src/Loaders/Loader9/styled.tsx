import styled from 'styled-components';
import { Props } from '../types';

export const Loader = styled.div<Props>`
    display: inline-block;
    width: 1em;
    height: 1em;
    color: ${pr=>pr.Color?pr.Color:'inherit'};
    vertical-align: middle;
    pointer-events: none;
    
    background-color: currentcolor;
	animation: loader-11 1.2s infinite ease-in-out;

    @keyframes loader-11 {
        0% {
            transform: perspective(120px) rotateX(0deg) rotateY(0deg);
        }
        50% {
            transform: perspective(120px) rotateX(-180.1deg) rotateY(0deg);
        }
        100% {
            transform: perspective(120px) rotateX(-180deg) rotateY(-179.9deg);
        }
    }

`;

const mixin = (size:any) => {
    switch (size) {
        case "XXS":
            return `font-size: 0.8rem;
           `;
        case "XS":
            return `font-size: 1rem;
            
            `;
        case "S":
            return `font-size: 2rem;
            `;
        
        case "M":
            return `font-size: 3rem;
            `;

        case "L":
            return `font-size: 5rem;
            `;
        
        case "XL":
            return `font-size: 6rem;
            `;
  
      default:
        return `font-size: 1rem;
       `;
     }
 };

export const LoaderWrapper = styled.div<Props>`
    ${pr => pr.disabled === true?'pointer-events: none;opacity: 0.4;':''}
    ${pr=>  mixin(pr.size)}
    display: ${pr=>pr.visible?'inline-block':'none'};
`;
