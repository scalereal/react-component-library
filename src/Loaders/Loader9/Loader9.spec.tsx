import { render } from '@testing-library/react';
import * as React from 'react';
import Loader9 from '.';

describe('Loader9 Loader', () => {
    it('should render with default theme', () => {
        const { container } = render(<     Loader9/>);
        expect(container).toMatchSnapshot();
    });
    it('should render with all the props given', () => {
        const { container } = render(
            <     Loader9
                size="M"
                Color='#126AFA'
            />
        );
        expect(container).toMatchSnapshot();
    });

    it('should render image avatar properly', () => {
        const { container } = render(
            <     Loader9 size="L" />
        );
        expect(container).toMatchSnapshot();
    });

});