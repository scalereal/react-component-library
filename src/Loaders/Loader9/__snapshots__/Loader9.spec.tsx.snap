// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader9 Loader should render image avatar properly 1`] = `
<div>
  <div
    class="sc-gsTCUz jVwPke"
  >
    <div
      class="sc-bdfBwQ dqCOSp"
    />
  </div>
</div>
`;

exports[`Loader9 Loader should render with all the props given 1`] = `
<div>
  <div
    class="sc-gsTCUz itwfOQ"
  >
    <div
      class="sc-bdfBwQ kDXqDg"
    />
  </div>
</div>
`;

exports[`Loader9 Loader should render with default theme 1`] = `
<div>
  <div
    class="sc-gsTCUz FEXmB"
  >
    <div
      class="sc-bdfBwQ dqCOSp"
    />
  </div>
</div>
`;
